### PR TITLE
Add return types and PHP generics

### PR DESCRIPTION
**Pull request by szepeviktor**

### CODE REVIEW

File: tests/Integration/RelationTest.php [hunk 1]
```diff
@@ -10,8 +10,9 @@
 class RelationTest extends BaseTest
 {
     /** @test */
-    public function basic_test()
+    public function basic_test(): void
     {
+        /** @var \Hammerstone\FastPaginate\Tests\Support\User $user */
         $user = User::first();
```
Comment (szepeviktor) [inline tests/Integration/RelationTest.php:16]: This could be `null`.

**Comment by aarondfrancis** [comment]
Thanks for the PR, but I don't want to add types to the package quite yet. I'm not personally a big fan of them.

**Comment by szepeviktor** [comment]
> I'm not personally a big fan of them.

All right.
Maybe later you develop a love-relation with typehints ❤️

**Comment by aarondfrancis** [comment]
> Maybe later you develop a love-relation with typehints ❤️

Maybe so! Don't give up on me yet 😉

**Pull request closed without merging**